cdap-securestore-ext-gcp-secretstore/src/main/java/io/cdap/cdap/securestore/gcp/cloudsecretmanager/WrappedSecret.java

```diff
@@ -0,0 +1,117 @@
+ * back.
+ *
+ * <p>Does not contain the underlying secret payload.
+ */
+public final class WrappedSecret {
+  private static final Gson GSON = new Gson();
+
+  private final String namespace;
+  private final SecretMetadata secretMetadata;
+
+  private WrappedSecret(String namespace, SecretMetadata secretMetadata) {
+    this.namespace = namespace;
+    this.secretMetadata = secretMetadata;
+  }
+
+  /** Constructs a new WrappedSecret from a CDAP Secret. */
+  public static WrappedSecret fromMetadata(String namespace, SecretMetadata metadata) {
+    return new WrappedSecret(namespace, metadata);
+  }
+
+  /** Constructs a new WrappedSecret from a GCP Secret Manager Secret. */
+  public static WrappedSecret fromGcpSecret(Secret secret) throws IOException {
+    String namespace = getNamespace(secret);
+    SecretMetadata metadata = toSecretMetadata(secret);
+    return new WrappedSecret(namespace, metadata);
+  }
+
+  public String getNamespace() {
+    return namespace;
+  }
+
+  public SecretMetadata getCdapSecretMetadata() {
+    return secretMetadata;
+  }
+
+  /**
+   * Returns a new GCP {@link Secret} representing the underlying SecretMetadata.
+   *
+   * @param resourceName Value to set for the "name" field needed for update operations.
+   */
+  public Secret getGcpSecret(String resourceName) {
+    return Secret.newBuilder()
+      // Set replication policy to automatic (as opposed to user-managed) and do not specify a
+      // CMEK (use google-managed key).
+      .setReplication(Replication.newBuilder().setAutomatic(Automatic.getDefaultInstance()))
+      .setName(resourceName)
+      .putAnnotations("cdap_namespace", namespace)
+      .putAnnotations("cdap_secret_name", secretMetadata.getName())
+      .putAnnotations(
+          "cdap_description", Optional.ofNullable(secretMetadata.getDescription()).orElse(""))
+      .putAnnotations("cdap_props", serializeProps(secretMetadata.getProperties()))
+      .build();
+  }
+
+  private static SecretMetadata toSecretMetadata(Secret secret) throws IOException {
+
+    return new SecretMetadata(
+      secret.getAnnotationsOrDefault("cdap_secret_name", ""),
+      secret.getAnnotationsOrDefault("cdap_description", ""),
+      Timestamps.toMillis(secret.getCreateTime()),
+      deserializeProps(secret.getAnnotationsOrDefault("cdap_props", "{}")));
+  }
+
+  private static String getNamespace(Secret secret) {
+    return secret.getAnnotationsOrDefault("cdap_namespace", "");
+  }
+
+  private static String serializeProps(Map<String, String> input) {
+    return GSON.toJson(input);
+  }
+
+  /**
+   * Parses {@code input} as JSON and returns the corresponding map, throwing an {@link IOException}
+   * if parsing input fails.
+   */
+  private static Map<String, String> deserializeProps(String input) throws IOException {
+    try {
+      return GSON.fromJson(input, new TypeToken<Map<String, String>>() { }.getType());
```

Comment:
this Type should be a private static final constant so we don't create it in each call